### ISSUE TITLE
notification stays longer than code is executing (MacOS)

### ISSUE THREAD

**Issue by VladimirFokow** [bhoov/manim-notebook#65]
## Log 1:

[Manim Notebook.log](https://github.com/user-attachments/files/17587019/Manim.Notebook.log)

## Video:

https://github.com/user-attachments/assets/7daeb310-aa35-42f2-b931-4cb3e4cd81bb

## Notes:

- (the last thing in the video) if I press almost immediately -> simultaneous commands aren't supported - that's OK
- if the interval is slightly longer but still very fast -> the code executes, but the notification stays for longer than code was executing (bad)

## Code:

```py
from manimlib import *

i = 0

class SquareToCircle(Scene):
    def construct(self):
        #
        ## Test
        i += 1
        print(i)
```

**Comment by Splines** [comment]
I think this is because VSCode has baked-in a minimal time that any notification should show, which is actually a good thing just for users to be able to at least read what the message is saying. Otherwise, imagine a notification popping up and the moment you move your eyes to it it's already gone...

You also notice this behavior when you start a log recording. While already started, the "Preparing everything..." notification stays a bit longer on screen.

Due to these reasons I will flag this as won't fix.

**Comment by VladimirFokow** [comment]
ok, but saying it's "executing" even though it's not - isn't clean

if anyone fixes this, I think it would be great

- the user doesn't need to read this notification every time. Only once - to know that it exists, and it can be read when user is executing some longer scene

**Comment by VladimirFokow** [comment]
(I'll leave it closed in case that there's really a hard limitation on VSCode's side that it's impossible to show a notification <1s in length)

**Comment by Splines** [comment]
Related are [these lines at vscode](https://github.com/microsoft/vscode/blob/b62b4161a0b5968f58242e602de2279230cf485b/src/vs/workbench/services/progress/browser/progressService.ts#L418-L427) and [this issue](https://github.com/microsoft/vscode/issues/93029). 

```js
// with a delay we only wait for the finish of the promise
if (typeof options.delay === 'number' && options.delay > 0) {
	await progressStateModel.promise;
}


// without a delay we show the notification for at least 800ms
// to reduce the chance of the notification flashing up and hiding
else {
	await Promise.all([timeout(800), progressStateModel.promise]);
}
```

But the `options.delay` is not accessible via the API as far as I know.
